pkg/controller/constraint/constraint_controller.go

```diff
@@ -323,10 +319,37 @@ func (r *ReconcileConstraint) Reconcile(ctx context.Context, request reconcile.R
 				if !isAPIEnabled {
 					r.log.V(1).Info("Warning: VAP API is not enabled, cannot create VAPBinding")
 					generateVAPB = false
-				}
-				if !HasVAPCel(ct) {
-					r.log.V(1).Info("Warning: ConstraintTemplate does not contain VAP-style CEL source, cannot create VAPBinding")
-					generateVAPB = false
+				} else {
+					ct := &v1beta1.ConstraintTemplate{}
+					err = r.reader.Get(ctx, types.NamespacedName{Name: strings.ToLower(instance.GetKind())}, ct)
+					if err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not get ConstraintTemplate for constraint")
+						}
+						return reconcile.Result{}, err
+					}
+
+					unversionedCT := &templates.ConstraintTemplate{}
+					if err := r.scheme.Convert(ct, unversionedCT, nil); err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not convert ConstraintTemplate to unversioned")
+						}
+						return reconcile.Result{}, err
+					}
+					hasVAP, err := ShouldGenerateVAP(unversionedCT)
+					if err != nil {
+						status.Status.Errors = append(status.Status.Errors, constraintstatusv1beta1.Error{Message: err.Error()})
+						if err2 := r.writer.Update(ctx, status); err2 != nil {
+							log.Error(err2, "could not determine if VAP exists to generate VAPB")
```

Comment:
This call doesn't check existence of VAP before generating VAPB. It only validates CT generateVap intent.

```suggestion
							log.Error(err2, "could not update constraint status error when determining if CT should generate VAP")
```